tools/cloud-build/daily-tests/ansible_playbooks/base-integration-test.yml

```diff
@@ -74,12 +74,32 @@
     - name: Print instance information
       ansible.builtin.debug:
         var: instances_list.stdout_lines
-    - name: Get remote IP
+    - name: Get IP of the remote node - Exact name provided
       changed_when: false
-      register: remote_ip
+      register: get_remote_ip
       ansible.builtin.command: >-
         gcloud compute instances describe --zone={{ zone }} {{ remote_node }}
         --format='get(networkInterfaces[0].accessConfigs[0].natIP)'
+      when: '"*" not in remote_node'
+    - name: Set remote_ip variable - Exact name provided
+      ansible.builtin.set_fact:
+        remote_ip: "{{ get_remote_ip.stdout }}"
+      when: '"*" not in remote_node'
+    - name: Get IP of the remote node - Name pattern provided
+      changed_when: false
+      register: get_remote_ip
+      ansible.builtin.command: >-
+        gcloud compute instances list \
+          --format='get(networkInterfaces[0].accessConfigs[0].natIP)' --limit=1 \
+          --filter=NAME:{{ remote_node }}
+      when: '"*" in remote_node'
+    - name: Set remote_ip variable - Name pattern provided
```

Comment:
Do you need this one twice (here and L84). Seems like you could do the two different gets and just set it once. And do we even need it at all, or could you just have each of the Gets set `remote_ip` directly?

Reply:
It looks like it. I can't recall why it was needed for the slurm tests, but from the looks of it, it might not have been. Updated, tests pending.

Reply:
I see now why it had to be defined twice. There is a variable and a "ansible fact". The variable is updated for any task that registers it, even if it's skipped, so I get this back when it's not a pattern (or the most recent task to register the variable):

```
TASK [Print remote node's public IP] *******************************************
ok: [localhost] => {
    "get_remote_ip.stdout": "VARIABLE IS NOT DEFINED!"
}
```

I missed it when testing Omnia because the most recent task to set the variable actually set it, but it failed the other base integration tests. I'm going to roll back to change and re-test.